src/templates/category-template.tsx

```diff
@@ -1,30 +1,42 @@
 import { graphql, Link } from "gatsby"
 import * as _ from "lodash"
 import * as React from "react"
+import ArchivesSort from "../components/ArchivesSort"
 import Layout from "../components/Layout"
 
 interface CategoriesTemplate {
   data: {
     allMarkdownRemark: {
-      distinct: string[]
+      nodes: [
+        {
+          frontmatter: {
+            date: string
+            author: string[]
+            title: string
+          }
+        }
+      ]
     }
   }
 }
 
 class CategoriesTemplate extends React.Component<CategoriesTemplate, {}> {
   render() {
-    const { data } = this.props
-    const allTitle = data.allMarkdownRemark.distinct
+    const { nodes } = this.props.data.allMarkdownRemark
 
     return (
       <div>
         <Layout>
-          {allTitle.map((singleTitle, index) => {
+          {nodes.map((singleNode, index) => {
+            const { title, author, date } = singleNode.frontmatter
             return (
               <div key={index}>
-                <Link to={`/blogs/${_.kebabCase(singleTitle)}`}>
-                  {singleTitle}
-                </Link>
+                <ArchivesSort
+                  currentPost={singleNode}
+                  preDate={
+                    index < 1 ? "empty" : nodes[index - 1].frontmatter.date
```

Comment:
Does the string "empty" have significance or could this be `null` as well?

Reply:
"Empty" just tells people who check the code when index < 1, it passes the "empty" to ArchivesSort component. I think I should change it to be ```null```.  Will change this. Thanks for the comment.